crates/biome_project/src/node_js_project/tsconfig_json.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{LanguageRoot, Manifest};
+use biome_deserialize::json::deserialize_from_json_ast;
+use biome_deserialize::{
+    Deserializable, DeserializableTypes, DeserializableValue, DeserializationDiagnostic,
+    DeserializationVisitor, Deserialized, Text,
+};
+use biome_json_syntax::JsonLanguage;
+use biome_text_size::TextRange;
+use rustc_hash::FxHashMap;
+
+#[derive(Debug, Default, Clone)]
+pub struct TsConfigJson {
+    base_url: Option<String>,
+    paths: FxHashMap<String, Vec<String>>,
+}
+
+impl Manifest for TsConfigJson {
+    type Language = JsonLanguage;
+
+    fn deserialize_manifest(root: &LanguageRoot<Self::Language>) -> Deserialized<Self> {
+        deserialize_from_json_ast::<TsConfigJson>(root, "")
+    }
+}
+
+impl Deserializable for TsConfigJson {
+    fn deserialize(
+        value: &impl DeserializableValue,
+        name: &str,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        value.deserialize(TsConfigJsonVisitor, name, diagnostics)
+    }
+}
+
+struct TsConfigJsonVisitor;
```

Comment:
Why not use the derive macro for implementing `Deserializable`?

Reply:
can you point to me the docs on how to use it?